src/main/java/com/aws/greengrass/lifecyclemanager/UnloadableService.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.lifecyclemanager;
+
+import com.aws.greengrass.config.Topic;
+import com.aws.greengrass.config.Topics;
+import com.aws.greengrass.lifecyclemanager.exceptions.ServiceLoadException;
+
+import java.util.Map;
+
+import static com.aws.greengrass.componentmanager.KernelConfigResolver.VERSION_CONFIG_KEY;
+import static com.aws.greengrass.deployment.bootstrap.BootstrapSuccessCode.REQUEST_RESTART;
+
+public class UnloadableService extends GreengrassService {
+    ServiceLoadException serviceLoadException;
+
+    public UnloadableService(Topics topics, ServiceLoadException e) {
+        super(topics);
+        this.serviceLoadException = e;
+    }
+
+    @Override
+    protected void install() {
+        serviceErrored(serviceLoadException);
+    }
+
+    @Override
+    public boolean isBootstrapRequired(Map<String, Object> newServiceConfig) {
```

Comment:
I'm still a little concerned about requiring nucleus restart for fixing unloadbale services, technically load doesn't require restart, this fix is just supposed to avoid breaking the nucleus completely if a service, generic or internal, happens to fail to load during restart. But for subsequent fixes to them, the deployment may not otherwise warrant a restart. Is it possible to avoid it, maybe by handling this during the service start/restart/remove handling and monitoring in deployment activator?

Reply:
Good point. Let me try that out